packages/ts/hilla-frontend/src/Connect.ts

```diff
@@ -247,26 +251,26 @@ export interface MiddlewareContext extends EndpointCallMetaInfo {
 /**
  * An async middleware callback that invokes the next middleware in the chain
  * or makes the actual request.
- * @param context The information about the call and request
+ * @param context - The information about the call and request
  */
-export type MiddlewareNext = (context: MiddlewareContext) => Promise<Response> | Response;
+export type MiddlewareNext = (context: MiddlewareContext) => MaybePromise<unknown>;
 
 /**
  * An interface that allows defining a middleware as a class.
  */
 export interface MiddlewareClass {
   /**
-   * @param context The information about the call and request
-   * @param next Invokes the next in the call chain
+   * @param context - The information about the call and request
+   * @param next - Invokes the next in the call chain
    */
-  invoke(context: MiddlewareContext, next: MiddlewareNext): Promise<Response> | Response;
+  invoke(context: MiddlewareContext, next?: MiddlewareNext): MaybePromise<unknown>;
 }
 
 /**
  * An async callback function that can intercept the request and response
  * of a call.
  */
-type MiddlewareFunction = (context: MiddlewareContext, next: MiddlewareNext) => Promise<Response> | Response;
+export type MiddlewareFunction = (context: MiddlewareContext, next?: MiddlewareNext) => MaybePromise<unknown>;
```

Comment:
Why `next` is optional? It is always provided for the middleware, and the middleware should normally invoke it.